P-NBLT/Crypto-Portfolio-Tracker
# Database set up

# Pull Request: Setup PostgreSQL Database Connection and Entity Definitions

## Overview
In this pull request, I have configured the connection to a PostgreSQL database hosted on a Docker container. To manage environment variables, `dotenv` has been installed and configured. Furthermore, two entities, `User` and `Holding`, have been defined to model the necessary data for our application.

### Key Changes:

1. **Environment Variables Management:**
   - Installed `dotenv` npm package.
   - Created a `.env` file to hold the necessary environment variables like `DB_HOST`, `DB_PORT`, `DB_USERNAME`, `DB_PASSWORD`, and `DB_DATABASE`.
   - Configured `dotenv` in the application to load environment variables.

2. **Entity Definitions:**
   - Created a `User` entity to model the users of the application. The entity has fields such as `id`, `firstName`, `lastName`, and a relation to the `Holding` entity.
   - Created a `Holding` entity to model the cryptocurrency holdings of the users. The entity has fields like `id`, `currency`, `amount`, `createdAt`, `lastUpdate`, and a relation to the `User` entity.

### How to Test:
1. Ensure Docker is running on your machine.
2. Run the following command to setup the database:
```bash
docker run --name <container-name> -e POSTGRES_USER=<username> -e POSTGRES_PASSWORD=<password> -e POSTGRES_DB=<db-name> -p 5432:5432 -d postgres
```
